enhancements/operator-conditions-lib.md

```diff
@@ -98,49 +157,17 @@ In this case, the specific API will return an error. The returned error can be w
 
 In this case, the error returned will be in the same format as stated in the previous scenario but would specifically mention that the requested object is not found.
 
-To summarize, these are the overall helpers present in the library. Here,
-- `*apiv1alpha1.Status` refers to the CR definition provided by OLM and
-- [`metav1.Condition`][metav1-Condition] refers to specific Condition type present in `status.conditions` array.
+### Advantage of using this format:
+1. Users need not import `operator-framework/api` to modify the conditions in the CR.
+2. Users can create a new interface which implements `ConditionManager` for custom conditions which are not supported by OLM.
 
-```go
-// GetNamespacedName - returns the NamespacedName of the CR.
-// Will return error when the namespace or name of the associated
-// Status CR cannot be found.
-GetNamespacedName() (types.NamespacedName, error)
-
-// SetConditionStatus - adds the specific condition to Status CR or updates the condition.Status
-// if the particular condition is already present.
-// Errors will be retured as specified in the `Scenarios` section.
-SetConditionStatus(status *apiv1alpha1.Status, condition metav1.Condition) error
-
-// RemoveConditionStatus - remove the specific condition present in Status CR.
-// Will return an error if the specified condition was not found in Status.
-RemoveConditionStatus(status *apiv1alpha1.Status, condition metav1.Condition) error
-
-// FindConditionStatus - finds the ConditionType in `status.conditions` array of Status CR.
-// Return nil if not present.
-FindConditionStatus(status *apiv1alpha1.Status, conditionType string) *metav1.Condition
-
-// IsConditionStatusTrue - returns true when the conditionType is present in `status.conditions`
-// and is set to `metav1.ConditionTrue`
-IsConditionStatusTrue(status *apiv1alpha1.Status, conditionType string) bool
-
-// IsStatusConditionPresentAndEqual - returns true if conditionType is present in
-// `status.conditions` and status is equal to as provided.
-IsConditionStatusPresentAndEqual(status *apiv1alpha1.Status, conditionType string, conditionStatus metav1.ConditionStatus) bool
-```
+### Disadvantages of using this format:
+1. In order to create/add a new condition, users will have to be directed to utilize the helpers provided in [apimachinery][apimachinery_condition]. 
```

Comment:
Do you think the slight tweak I suggested above about `condition` and `Condition` makes this simpler? New condition types could be supported via our `Conditions` manager, and a new constructor that returns a `Condition` for the new type.

We could add a `NewCondition()` function to support this:

```go
func NewCondition(cl client.Client, conditionType metav1.ConditionType) (Condition, error) {
	c, err := NewConditions(cl)
	if err != nil {
		return nil, err
	}
	return &condition{conditions: c, conditionType: conditionType}, nil
}
```

And then `NewUpgradable()` would simplify to:

```go
func NewUpgradable(cl client.Client) (Condition, error) {
	return NewCondition(cl, "Upgradable")
}
```